ms2query/ms2library.py

```diff
@@ -0,0 +1,332 @@
+
+        # Set given settings
+        self.spectrum_id_column_name = settings["spectrum_id_column_name"]
+        # todo create a ms2query model class that stores the model but also the
+        #  settings used, since the settings used should always be the same as
+        #  when the model was trained
+        self.cosine_score_tolerance = settings["cosine_score_tolerance"]
+        self.base_nr_mass_similarity = settings["base_nr_mass_similarity"]
+        # todo make new model that has a fixed basic mass
+        self.max_parent_mass = settings["max_parent_mass"]
+
+        # Load models and set sqlite_file_location
+        self.sqlite_file_location = sqlite_file_location
+        self.s2v_model = Word2Vec.load(s2v_model_file_name)
+        self.ms2ds_model = load_ms2ds_model(ms2ds_model_file_name)
+
+        # loads the library embeddings into memory
+        self.s2v_embeddings: pd.DataFrame = load_pickled_file(
+            pickled_s2v_embeddings_file_name)
+        self.ms2ds_embeddings: pd.DataFrame = load_pickled_file(
+            pickled_ms2ds_embeddings_file_name)
+
+    @staticmethod
+    def _set_settings(new_settings: Dict[str, Union[str, bool]],
+                      ) -> Dict[str, Union[str, float]]:
+        """Changes default settings to new_settings and creates file names
+
+        Args:
+        ------
+        new_settings:
+            Dictionary with settings that should be changed. Only the
+            keys given in default_settings can be used and the type has to be
+            the same as the type of the values in default settings.
+        """
+        # Set default settings
+        default_settings = {"spectrum_id_column_name": "spectrumid",
+                            "cosine_score_tolerance": 0.1,
+                            "base_nr_mass_similarity": 0.8,
+                            "max_parent_mass": 13418.370894192036}
+
+        for attribute in new_settings:
+            assert attribute in default_settings, \
+                f"Invalid argument in constructor:{attribute}"
+            assert isinstance(new_settings[attribute],
+                              type(default_settings[attribute])), \
+                f"Different type is expected for argument: {attribute}"
+            default_settings[attribute] = new_settings[attribute]
+        return default_settings
+
+    def select_best_matches(self,
+                            query_spectra: List[Spectrum],
+                            ms2query_model_file_name: str
+                            ) -> Dict[str, pd.DataFrame]:
+        """Returns ordered best matches with info for all query spectra
+
+        Args
+        ----
+        query_spectra:
+            List of query spectra for which the best matches should be found
+        ms2query_model_file_name:
+            File name of a hdf5 file containing the ms2query model.
+        """
+        # Selects top 20 best matches based on ms2ds and calculates all scores
```

Comment:
20 is an arbitraty value here. Where can it be set?

Reply:
It can already be set in pre_select_spectra (where the default = 20), I might want to make it possible to set this already in settings of the MS2Library object. For now I leave it like this, since there will be quite some changes to the preselection, when splitting it into two different workflows, once I implemented it correctly, I will make sure that the settings for preselection can be changed in a suitable way.